### PR TITLE
Add a setting for parts exposing network capabilities

### PR DESCRIPTION
**Pull request by josephcsible**

### CODE REVIEW

File: src/main/java/org/cyclops/integrateddynamics/api/part/IPartState.java [hunk 1]
```diff
@@ -78,6 +78,17 @@
...
+    /**
+     * Set whether this part should expose network capabilities.
+     * @param exposeCapabilities Whether this part should expose network capabilities.
+     */
+    public void setExposeCapabilities(boolean exposeCapabilities);
+
+    /**
+     * @return Whether this part should expose network capabilities.
+     */
+    public boolean getExposeCapabilities();
```
Comment (rubensworks) [inline src/main/java/org/cyclops/integrateddynamics/api/part/IPartState.java]: `isExposeCapabilities `

File: src/main/java/org/cyclops/integrateddynamics/api/part/IPartType.java [hunk 2]
```diff
@@ -133,6 +133,19 @@
...
+     * @param state The state
+     * @param exposeCapabilities Whether to expose network capabilities.
+     */
+    public void setExposeCapabilities(S state, boolean exposeCapabilities);
+
+    /**
+     * @param state The state
+     * @return Whether this part should expose network capabilities.
+     */
+    public boolean getExposeCapabilities(S state);
```
Comment (rubensworks) [inline src/main/java/org/cyclops/integrateddynamics/api/part/IPartType.java]: `isExposeCapabilities `

File: src/main/java/org/cyclops/integrateddynamics/core/client/gui/container/GuiPartSettings.java [hunk 3]
```diff
@@ -145,6 +157,16 @@ protected int getBaseXSize() {
...
     }
 
+    @Override
+    protected int getBaseYSize() {
+        return 189;
+    }
+
+    protected void updateExposeCapabilities(boolean newValue) {
+        exposeCapabilities = newValue;
+        buttonExposeCapabilities.displayString = newValue ? trueStr : falseStr;
```
Comment (rubensworks) [inline src/main/java/org/cyclops/integrateddynamics/core/client/gui/container/GuiPartSettings.java]: I don't really like buttons that toggle that update their label like this. A checkbox or an actual visual toggle-able button would be better. (perhaps similar to how the new mechanical machines have a toggle to auto-export fluids)

File: src/main/java/org/cyclops/integrateddynamics/core/part/PartStateBase.java [hunk 4]
```diff
@@ -217,15 +230,17 @@ public void gatherCapabilities(P partType) {
 
     @Override
     public boolean hasCapability(Capability<?> capability) {
-        return volatileCapabilities.containsKey(capability)
+        return (exposeCapabilities && volatileCapabilities.containsKey(capability))
```
Comment (rubensworks) [inline src/main/java/org/cyclops/integrateddynamics/core/part/PartStateBase.java]: Call getter of `exposeCapabilities`.

File: src/main/java/org/cyclops/integrateddynamics/core/part/PartStateBase.java [hunk 5]
```diff
@@ -217,15 +230,17 @@ public void gatherCapabilities(P partType) {
...
                 || (capabilities != null && capabilities.hasCapability(capability, null));
     }
 
     @Override
     public <T> T getCapability(Capability<T> capability) {
-        Object o = volatileCapabilities.get(capability);
-        if(o != null) {
-            return (T) o;
+        if(exposeCapabilities) {
```
Comment (rubensworks) [inline src/main/java/org/cyclops/integrateddynamics/core/part/PartStateBase.java]: Call getter of `exposeCapabilities`.

**Review by rubensworks — CHANGES_REQUESTED** [review]
Some small comments. My main issue is with how the toggle button is implemented.

**Comment by josephcsible** [comment]
Holy merge conflict, Batman!

File: src/main/java/org/cyclops/integrateddynamics/core/client/gui/container/GuiPartSettings.java (hunk 3 above)
Comment (josephcsible) [reply src/main/java/org/cyclops/integrateddynamics/core/client/gui/container/GuiPartSettings.java]: I don't see a way to make the auto-export button style not look weird here. And isn't this how the in-game config buttons work?

Comment (rubensworks) [reply src/main/java/org/cyclops/integrateddynamics/core/client/gui/container/GuiPartSettings.java]: Something based on a checkbox should be doable, possibly using an image-based button using that green checkmark in `Images`.

**Comment by rubensworks** [comment]
This will be tackled differently, as discussed in private.

**Pull request closed without merging**